torch/_subclasses/fake_utils.py

```diff
@@ -50,16 +87,41 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
                     fake_args, fake_kwargs = pytree.tree_map_only(
                         torch.Tensor, fake_mode.from_tensor, (args, kwargs)
                     )
-                    fake_r = func(*fake_args, **fake_kwargs)
+                    with warnings.catch_warnings():
+                        fake_r = func(*fake_args, **fake_kwargs)
             except UnsupportedFakeTensorException:
                 pass
 
         r = func(*args, **kwargs)
         if fake_r is not None:
+            r_flat = tree_flatten(r)[0]
+            f_flat = tree_flatten(fake_r)[0]
+            assert len(r_flat) == len(
+                r_flat
+            ), f"Mismatch {len(r_flat)} != {len(r_flat)} on {func}"
```

Comment:
technically you should check the specs too

Reply:
Filed issue https://github.com/pytorch/pytorch/issues/85438